src/pycatcher/catch.py

```diff
@@ -513,9 +513,9 @@ def generate_outliers_stl(df, type, period) -> pd.DataFrame:
 
     if type == 'additive':
         logging.info("Outlier detection using STL Additive Model")
-        stl = STL(df, period=period)
+        df_add = df.copy()
```

Comment:
Just curious to know why do we need a new clone?

Reply:
For better readability/reference about different logic for additive and multiplicative seasonalities :)

Reply:
Ah! Ok. I need to check on the memory usage when doing some performance testing.